spec/activityDetection.spec.ts

```diff
@@ -0,0 +1,66 @@
+describe('activity detection', () => {
+
+    const originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;
+
+    beforeEach(() => {
+        jest.resetModules();
+        jest.useFakeTimers();
+    });
+
+    afterAll((done) => {
+        jasmine.DEFAULT_TIMEOUT_INTERVAL = originalTimeout;
+        done();
+    });
+
+    it('should call `setWindowAndThreshold` correctly', () => {
+        const { activityDetection } = require('../src/app/activity-detection');
+
+        // mocking startActivityMonitor
+        jest.spyOn(activityDetection, 'startActivityMonitor')
+        .mockImplementation(() => jest.fn());
+
+        const spy = jest.spyOn(activityDetection, 'setWindowAndThreshold');
+
+        activityDetection.setWindowAndThreshold({}, 1000);
+
+        expect(spy).toBeCalledWith({}, 1000);
+    });
+
+    it('should start activity monitor when `setWindowAndThreshold` is called', () => {
+        const { activityDetection } = require('../src/app/activity-detection');
+
+        const spy = jest.spyOn(activityDetection, 'startActivityMonitor')
+        .mockImplementation(() => jest.fn());
+
+        activityDetection.setWindowAndThreshold({}, 1000);
+
+        expect(spy).toBeCalled();
+    });
+
+    it('should call `activity` when `startActivityMonitor` is called', () => {
+        const { activityDetection } = require('../src/app/activity-detection');
```

Comment:
any reason we are requiring here? Can't we use import at the beginning of the file?